### PR TITLE
eden config remote-delete, refactor

### PR DESCRIPTION
**Pull request by prog893**
1. Implemented `eden config remote-delete` command for removal of remote profiles.
2. `cmdline.py` now only has main, parser builder and parser handlers. Most of other methods go to `utils.py`
3. config file parsing refactored into methods
4. Restructure parser: config command now has it's own subparsers for easier command addition (more to be added soon). 

TODO:
- [x] fix README

### CODE REVIEW

File: aws_eden_cli/cmdline.py [hunk 1]
```diff
@@ -58,6 +72,11 @@ def create_parser():
 
         i.add_argument('-v', '--verbose', action='store_true')
 
+    # switches for remote subcommands
+    for i in [parser_config_push, ]:
+        i.add_argument('--remote-table-name', type=str, required=False, default='eden',
+                       help='profile name in eden configuration file')
```
Comment (toVersus) [inline aws_eden_cli/cmdline.py]: Cloud you update a help message for this arg? Also, update the Usage section in README.

File: aws_eden_cli/utils.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            errors += 1
+            continue
+    return errors
+
+
+def create_envvar_dict(args, config):
+    variables = {}
+    profile_name = args['profile']
+
+    for p in consts.parameters:
```
Comment (toVersus) [inline aws_eden_cli/utils.py]: I found orthographical variant here:

```suggestion
    for parameter in consts.parameters:
```

File: aws_eden_cli/utils.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        logger.debug("Skipping ConfigParser DEFAULT profile (comes up even if not in file)")
+        return 0
+
+    if profile not in config:
+        logger.error(f"Profile {profile} is not in config file")
+        errors += 1
+        return errors
+
+    for parameter in consts.parameters:
+        key = parameter['name']
```
Comment (toVersus) [inline aws_eden_cli/utils.py:66]: `parameter_name` is used for variable name in `create_envvar_dict` function. I prefer that one.

File: aws_eden_cli/dynamodb.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+import logging
```
Comment (toVersus) [inline aws_eden_cli/dynamodb.py:1]: If possible, could you move below lines to this file as well?

https://github.com/baikonur-oss/aws-eden-cli/blob/6299497b2f90ecbb2787a6f3aee474d3bf64b0f2/aws_eden_cli/cmdline.py#L137-L150
https://github.com/baikonur-oss/aws-eden-cli/blob/6299497b2f90ecbb2787a6f3aee474d3bf64b0f2/aws_eden_cli/cmdline.py#L161-L173

File: aws_eden_cli/utils.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+def config_write_overrides(config, args):
+    updated = False
+
+    profile_name = args['profile']
+
+    if profile_name not in config:
+        config[profile_name] = {}
+
+    for parameter in consts.parameters:
+        key = parameter['name']
```
Comment (toVersus) [inline aws_eden_cli/utils.py:38]: `parameter_name` is used for variable name in `create_envvar_dict` function. I prefer that one.

File: aws_eden_cli/utils.py (hunk 3 above)
Comment (prog893) [reply aws_eden_cli/utils.py:66]: This one is different from others. parameter variable value changes on every loop iteration, thus can not be parametrized.

File: aws_eden_cli/utils.py (hunk 5 above)
Comment (prog893) [reply aws_eden_cli/utils.py:38]: This one changes on every iteration as well

**Review by toVersus — APPROVED** [review]
LGTM

**Pull request merged**